torch/csrc/autograd/functions/accumulate_grad.h

```diff
@@ -7,11 +7,12 @@
 namespace torch { namespace autograd {
 
 struct TORCH_API AccumulateGrad : public Node {
-  explicit AccumulateGrad(Variable variable_);
+  explicit AccumulateGrad(const Variable& variable_);
 
   variable_list apply(variable_list&& grads) override;
 
-  Variable variable;
+  WeakVariable variable;
+  WeakVariable variable_grad;
```

Comment:
About here would be a good explanation about why it's OK for this to be weak. (Not too hard to understand: if the variable is dead, no need to accumulate into it.) But why also record `variable_grad`? That would be a good thing to explain in line here.

Reply:
Yes, definitely, I'll add more comments (also in other places) when the `detach()` question is settled.